### PR TITLE
Remapping Validity Check Feature

### PR DESCRIPTION
**Pull request by mlodi-hqs**
* Added `unstable_remapping_validity_check` feature

### CODE REVIEW

**Comment by mlodi-hqs** [comment]
Now updated to `alpha.2`

File: roqoqo/src/operations/pragma_operations.rs
Comment (kbarkhqs) [inline roqoqo/src/operations/pragma_operations.rs:1]: Could you add tests for this behaviour with the feature? (for PragmaStartDecompositionBlock, PragmaControlledCircuit and PragmaChangeDevice)

File: roqoqo/src/operations/analog_operations.rs
Comment (kbarkhqs) [inline roqoqo/src/operations/analog_operations.rs:1]: Also here, I would suggesting adding tests

File: roqoqo/src/operations/measurement_operations.rs
Comment (kbarkhqs) [inline roqoqo/src/operations/measurement_operations.rs:1]: Same here, I would suggest adding tests to check the behaviour is correct

File: roqoqo-derive/src/substitute.rs
Comment (kbarkhqs) [inline roqoqo-derive/src/substitute.rs:1]: Could you add a test for one of the structs that derives this, to check it?

File: roqoqo-derive/src/substitute_modes.rs
Comment (kbarkhqs) [inline roqoqo-derive/src/substitute_modes.rs:1]: Could you add a test for one of the structs that derives this, to check it?

**Review by kbarkhqs — CHANGES_REQUESTED** [review]
Further unittests for completeness would be helpful

**Comment by mlodi-hqs** [comment]
The runners will only run `cargo test --no-default features`, but locally `cargo test --no-defeault-features --features unstable_remapping_validity_check ` works too.

**Review by kbarkhqs — APPROVED** [review]

**Comment by mlodi-hqs** [comment]
No longer necessary as I got around the problem in another way.

**Pull request closed without merging**